pkg/controllers/workgenerator/controller.go

```diff
@@ -558,17 +566,17 @@ func (r *Reconciler) syncAllWork(ctx context.Context, resourceBinding *fleetv1be
 				}
```

Comment:
one improvement is in 562, instead of triggering deleting, could you please check if the work has deletion timestamp already? if not, we call delete. If not, log the work and its deletion timestamp for debugging purpose.

set the updateAny as true to indicate the deletion is still in progressing